[2025-04-08T20:04:08.553+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-08T20:04:08.579+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Download_Stock_Price.merge_stock_price manual__2025-04-08T20:03:49.932668+00:00 [queued]>
[2025-04-08T20:04:08.592+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Download_Stock_Price.merge_stock_price manual__2025-04-08T20:03:49.932668+00:00 [queued]>
[2025-04-08T20:04:08.592+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 2
[2025-04-08T20:04:08.612+0000] {taskinstance.py:2890} INFO - Executing <Task(SQLExecuteQueryOperator): merge_stock_price> on 2025-04-08 20:03:49.932668+00:00
[2025-04-08T20:04:08.620+0000] {standard_task_runner.py:72} INFO - Started process 27908 to run task
[2025-04-08T20:04:08.625+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'Download_Stock_Price', 'merge_stock_price', 'manual__2025-04-08T20:03:49.932668+00:00', '--job-id', '36', '--raw', '--subdir', 'DAGS_FOLDER/download_stock_price.py', '--cfg-path', '/tmp/tmp84fuy_ll']
[2025-04-08T20:04:08.627+0000] {standard_task_runner.py:105} INFO - Job 36: Subtask merge_stock_price
[2025-04-08T20:04:08.698+0000] {task_command.py:467} INFO - Running <TaskInstance: Download_Stock_Price.merge_stock_price manual__2025-04-08T20:03:49.932668+00:00 [running]> on host instance-20250307-222520.us-central1-a.c.esoteric-realm-448604-v0.internal
[2025-04-08T20:04:08.805+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='airflow@example.com' AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='Download_Stock_Price' AIRFLOW_CTX_TASK_ID='merge_stock_price' AIRFLOW_CTX_EXECUTION_DATE='2025-04-08T20:03:49.932668+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-08T20:03:49.932668+00:00'
[2025-04-08T20:04:08.807+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-04-08T20:04:08.807+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-04-08T20:04:08.807+0000] {logging_mixin.py:190} INFO - Current task name:merge_stock_price state:running start_date:2025-04-08 20:04:08.579461+00:00
[2025-04-08T20:04:08.808+0000] {logging_mixin.py:190} INFO - Dag name:Download_Stock_Price and current dag run status:running
[2025-04-08T20:04:08.808+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-04-08T20:04:08.809+0000] {sql.py:296} INFO - Executing: -- update the existing rows 
update stock_prices p, stock_prices_stage s 
set p.open_price = s.open_price, 
    p.high_price = s.high_price, 
    p.low_price = s.low_price, 
    p.close_price = s.close_price, 
    update_at = now() 
where p.ticker = s.ticker 
and p.as_of_date = s.as_of_date 

-- inserting new rows 
insert into stock_prices 
(ticker, as_of_date, open_price, high_price, low_price, close_price) 
select ticker, as_of_date, open_price, high_price, low_price, close_price 
from stock_prices_stage s 
where not exists 
(select 1 from stock_prices p 
where p.ticker = s.ticker 
and p.as_of_date = s.as_of_date); 

-- truncate the stage table; 
truncate table stock_prices_stage; 

[2025-04-08T20:04:08.821+0000] {base.py:84} INFO - Retrieving connection 'demodb'
[2025-04-08T20:04:08.836+0000] {base.py:84} INFO - Retrieving connection 'demodb'
[2025-04-08T20:04:08.842+0000] {sql.py:624} INFO - Running statement: -- update the existing rows 
update stock_prices p, stock_prices_stage s 
set p.open_price = s.open_price, 
    p.high_price = s.high_price, 
    p.low_price = s.low_price, 
    p.close_price = s.close_price, 
    update_at = now() 
where p.ticker = s.ticker 
and p.as_of_date = s.as_of_date 

-- inserting new rows 
insert into stock_prices 
(ticker, as_of_date, open_price, high_price, low_price, close_price) 
select ticker, as_of_date, open_price, high_price, low_price, close_price 
from stock_prices_stage s 
where not exists 
(select 1 from stock_prices p 
where p.ticker = s.ticker 
and p.as_of_date = s.as_of_date); 

-- truncate the stage table; 
truncate table stock_prices_stage; 
, parameters: None
[2025-04-08T20:04:08.843+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/providers/common/sql/operators/sql.py", line 302, in execute
    output = hook.run(
             ^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 580, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 629, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.SyntaxError: syntax error at or near ","
LINE 2: update stock_prices p, stock_prices_stage s 
                             ^

[2025-04-08T20:04:08.866+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-04-08T20:04:08.866+0000] {logging_mixin.py:190} INFO - Task start:2025-04-08 20:04:08.579461+00:00 end:2025-04-08 20:04:08.865705+00:00 duration:0.286244
[2025-04-08T20:04:08.867+0000] {logging_mixin.py:190} INFO - Task:<Task(SQLExecuteQueryOperator): merge_stock_price> dag:<DAG: Download_Stock_Price> dagrun:<DagRun Download_Stock_Price @ 2025-04-08 20:03:49.932668+00:00: manual__2025-04-08T20:03:49.932668+00:00, state:running, queued_at: 2025-04-08 20:03:49.951100+00:00. externally triggered: True>
[2025-04-08T20:04:08.867+0000] {logging_mixin.py:190} INFO - Failure caused by syntax error at or near ","
LINE 2: update stock_prices p, stock_prices_stage s 
                             ^
[2025-04-08T20:04:08.867+0000] {taskinstance.py:1226} INFO - Marking task as UP_FOR_RETRY. dag_id=Download_Stock_Price, task_id=merge_stock_price, run_id=manual__2025-04-08T20:03:49.932668+00:00, execution_date=20250408T200349, start_date=20250408T200408, end_date=20250408T200408
[2025-04-08T20:04:08.932+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-04-08T20:04:08.932+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 36 for task merge_stock_price (syntax error at or near ","
LINE 2: update stock_prices p, stock_prices_stage s 
                             ^
; 27908)
Traceback (most recent call last):
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/providers/common/sql/operators/sql.py", line 302, in execute
    output = hook.run(
             ^^^^^^^^^
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 580, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/gloria/airflow_env/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 629, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.SyntaxError: syntax error at or near ","
LINE 2: update stock_prices p, stock_prices_stage s 
                             ^

[2025-04-08T20:04:08.960+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-04-08T20:04:08.994+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-08T20:04:08.997+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
